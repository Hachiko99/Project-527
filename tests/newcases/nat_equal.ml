#use "nat.decls"

let nat_eq : nat -> nat -> bool |>
{
  0 => ( 0 => True
       | 1 => False )
| 1 => ( 0 => False
       | 1 => True
       | 2 => False )
} = ?
