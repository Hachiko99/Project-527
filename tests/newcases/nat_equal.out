(* Generated Tue Apr 19 12:38:12 2016 *)
let nat_eq : nat -> nat -> bool =
  let rec f1 (n1:nat) : nat -> bool =
    fun (n2:nat) ->
      match n1 with
        | O -> (match n2 with
                  | O -> True
                  | S (n3) -> False)
        | S (n3) -> (match n2 with
                       | O -> False
                       | S (n4) -> f1 n3 n4)
  in
    f1
;;
